src/dotnet-roslyn/Utilities/AzDOConnection.cs

```diff
@@ -18,20 +18,24 @@ internal sealed class AzDOConnection : IDisposable
     private bool _disposed = false;
 
     public string BuildProjectName { get; }
+    public string BuildDefinitionName { get; }
     private VssConnection Connection { get; }
     public GitHttpClient GitClient { get; }
     public BuildHttpClient BuildClient { get; }
+    public HttpClient NuGetClient { get; }
     public FileContainerHttpClient ContainerClient { get; }
     public ProjectHttpClient ProjectClient { get; }
 
-    public AzDOConnection(string azdoUrl, string projectName, SecretClient client, string secretName)
+    public AzDOConnection(string azdoUrl, string projectName, string buildDefinitionName, SecretClient client, string secretName)
```

Comment:
I hate `buildDefinitionName` 😋

Reply:
Yeah, I dunno if BuildDefinition is an old TFS-ism or not. `BuildPipelineName` or `PipelineName` are likely more meaningful to us. CreateTags uses it and I wanted to unify the AzDOConnection classes. Happy to take changes in a follow up.

Reply:
Yeah, I also wanted to unify the AzDOConnection classes in https://github.com/dotnet/roslyn-tools/pull/1189 but the build pipeline name is just annoying, because its AzDo project specific, and product specific. Will update that PR now that this is merged though, and see where we land.

Reply:
The real funs begins when we move RIT to be part of dotnet-roslyn (https://github.com/dotnet/roslyn-tools/issues/1168)